### DISCUSSION TITLE
PP-OCRv3开启训练过程中，报错：RecursionError: maximum recursion depth exceeded

### DISCUSSION THREAD

**Discussion by pcycccccc** [PaddlePaddle/PaddleOCR#12742] [Q&A] [answered]
- 数据集：一个两百多张的图像，使用PPOCRLabel进行标注保存标签，后使用PPOCRLabel中的gen_ocr_train_val_test.py进行划分数据集。
可以看到数据集的格式如下，我感觉是没有什么问题：
![image](https://github.com/PaddlePaddle/PaddleOCR/assets/108455796/ff177eb6-f32f-49e3-b7af-981af7f43c6d)

- 预训练模型：下载en_PP-OCRv3_rec_train

- 修改了configs\rec\PP-OCRv3\en_PP-OCRv3_rec.yml文件中的文件路径（也就是改成绝对路径而已）
![image](https://github.com/PaddlePaddle/PaddleOCR/assets/108455796/987b56b4-3632-4489-98ef-d39c5baef6aa)

还修改了其中的data路径，修改了num_workers: 2，其他没动
![image](https://github.com/PaddlePaddle/PaddleOCR/assets/108455796/715f06c1-49ed-43c9-8f88-ca8e483538ff)
![image](https://github.com/PaddlePaddle/PaddleOCR/assets/108455796/e1c4d2e7-0ac9-4cc2-b4fc-b9660bb431b1)

- 开启训练的指令（开启GPU来训练）：
`python tools/train.py -c configs/rec/PP-OCRv3/en_PP-OCRv3_rec.yml -o Global.pretrained_model=./pretrain_models/en_PP-OCRv3_rec_train/best_accuracy`

- 输出报错的信息（卡在ppocr INFO: During the training process, after the 0th iteration, an evaluation is run every 2000 iterations这里很长一段时间后，就会出现下面的错误）：
![image](https://github.com/PaddlePaddle/PaddleOCR/assets/108455796/e0a3229e-32f2-4026-8ad5-d5e6b654239f)

有没有大佬可以看一下，为啥会出现这种问题，我该如何去解决呢？

**Comment by GreatV** [comment]
提供的 [公开数据集](https://github.com/PaddlePaddle/PaddleOCR/blob/main/doc/doc_ch/dataset/ocr_datasets.md#12-%E5%85%AC%E5%BC%80%E6%95%B0%E6%8D%AE%E9%9B%86) 可以训练吗？

**Comment by pcycccccc** [comment, marked as answer]
@GreatV  感谢您的回复，我今天下载了公开数据集文字识别的文件，并进行了训练，现在已经能正常跑了。另外注意到，我原先执行训练遇到的问题可能是因为我数据集路径弄错了，我需要训练的文字识别模型，但是我提供的数据集路径是关于文字检测的（det/train.txt），在我更新数据集后，使用rec/train.txt，能够正常训练了。

**Reply by kishore-work-hard** [reply]
> @GreatV 感谢您的回复，我今天下载了公开数据集文字识别的文件，并进行了训练，现在已经能正常跑了。另外注意到，我原先执行训练遇到的问题可能是因为我数据集路径弄错了，我需要训练的文字识别模型，但是我提供的数据集路径是关于文字检测的（det/train.txt），在我更新数据集后，使用rec/train.txt，能够正常训练了。

bro please type in English.
